metricflow/plan_conversion/node_processor.py

```diff
@@ -217,6 +215,8 @@ def _get_candidates_nodes_for_multi_hop(
                     left_instance_set=data_set_of_first_node_that_could_be_joined.instance_set,
                     right_instance_set=data_set_of_second_node_that_can_be_joined.instance_set,
                     on_entity_reference=entity_reference_to_join_first_and_second_nodes,
+                    # TODO: make this check more substantial in V2
+                    right_node_is_subquery=isinstance(second_node_that_could_be_joined, ComputeMetricsNode),
```

Comment:
Yeah, I thought this might be controversial 😅 
I guess I don't love this solution, so I'm trying to think of a better one. What if there is another node that technically is aggregated? We probably do have other nodes that are aggregated - like in `FilterElementsNode` we sometimes aggregate to the distinct values.

Reply:
Ok I'm thinking instead of `BaseOutput.is_aggregated -> bool`, I'll add something like `BaseOutput.is_aggregated_to_elements -> Sequence[LinkableElementSpec]`. That way we can store the list of entities & dimensions it's aggregated to, if possible. Should be useful for V2 of this feature, too.

Reply:
Ok I added that property! Took a few new commits so you can start with https://github.com/dbt-labs/metricflow/pull/1169/commits/8951ea06c64918ccfff060ad42345b45b9ded249. I'm pretty happy with this change, makes the join checks much more secure 👍